### PR TITLE
we don't need to check permissions twice

### PR DESCRIPTION
**Pull request by schiessle**
we check the permission already here: https://github.com/nextcloud/server/compare/cleanup-after-sync?expand=1#diff-efc44835a2c59b3707c965b80a3d669eL338

cc @LukasReschke @MorrisJobke

### CODE REVIEW

**Comment by LukasReschke** [comment]
👍

File: apps/files_versions/lib/Storage.php [hunk 1]
```diff
@@ -320,16 +320,10 @@ public static function rollback($file, $revision) {
 			// add expected leading slash
 			$file = '/' . ltrim($file, '/');
 			list($uid, $filename) = self::getUidAndFilename($file);
-			if ($uid === null || trim($filename, '/') === '') {
```
Comment (icewind1991) [inline apps/files_versions/lib/Storage.php]: I'm pretty sure this is needed to handle some edge cases but I don't remember the details

Comment (schiessle) [inline apps/files_versions/lib/Storage.php]: I can't think of any, but I will put it back... can't hurt

**Comment by schiessle** [comment]
@icewind1991  I added the additional check again, please review. Thanks!

**Comment by LukasReschke** [comment]
> @icewind1991 I added the additional check again, please review. Thanks!

Can you backport the additional check also to stable9?

**Comment by schiessle** [comment]
@LukasReschke done: https://github.com/nextcloud/server/pull/304

**Comment by icewind1991** [comment]
:+1:

**Comment by MorrisJobke** [comment]
👍

**Pull request merged**